### PR TITLE
Introduce StatsPredicates

### PR DESCRIPTION
**Pull request by JingsongLi**

### CODE REVIEW

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/eval/NotEqual.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public Boolean eval(Object[] values) {
+        Object field = values[index];
+        return field == null ? null : literal.compareValueTo(field) != 0;
+    }
+
+    @Override
+    public boolean eval(long rowCount, FieldStats[] fieldStats) {
+        // possible hit
+        return true;
```
Comment (tsreaper) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/eval/NotEqual.java:49]: If `minValue == maxValue` and `minValue == literal` then this is definitely false.

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/eval/IsNotNull.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Nullable
+    @Override
+    public Boolean eval(Object[] values) {
+        return values[index] != null;
+    }
+
+    @Override
+    public boolean eval(long rowCount, FieldStats[] fieldStats) {
+        return fieldStats[index].nullCount() == 0;
```
Comment (tsreaper) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/eval/IsNotNull.java:42]:
```suggestion
        return fieldStats[index].nullCount() != rowCount;
```

Comment (JingsongLi) [reply dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/eval/IsNotNull.java]: It is a bug...

**Comment by JingsongLi** [comment]
Moved to new repo.

**Pull request closed without merging**